great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -258,16 +258,23 @@ def get_batch_list_from_batch_request(
                 )
             # Creating the batch_spec is our hook into the execution engine.
             batch_spec = SqlAlchemyDatasourceBatchSpec(**batch_spec_kwargs)
-            data, markers = self.datasource.execution_engine.get_batch_data_and_markers(
-                batch_spec=batch_spec
-            )
 
             # batch_definition (along with batch_spec and markers) is only here to satisfy a
             # legacy constraint when computing usage statistics in a validator. We hope to remove
             # it in the future.
             # imports are done inline to prevent a circular dependency with core/batch.py
             from great_expectations.core import IDDict
-            from great_expectations.core.batch import BatchDefinition
+            from great_expectations.core.batch import (
```

Comment:
Can you make a function like this:
```
def update_batch_forward_refs():
    from great_expectations.core.batch import (
                BatchData,
                BatchDefinition,
                BatchMarkers,
            )
      Batch.update_forward_refs(
                BatchData=BatchData,
                BatchDefinition=BatchDefinition,
                BatchMarkers=BatchMarkers,
            )
```
and put it in a common location so we can call it here and in `pandas_datasource`? If you need some of these types for type checking you can put it in a `TYPE_CHECKING` block.

Reply:
I was able to make this an overloaded `classmethod` that imports within the method and then passes the kwargs to `super()`! Definitely something we should do if we need postponed annotations in the future.